sdk/python/kfp/_client.py

```diff
@@ -155,9 +155,10 @@ def _load_config(self, host, client_id, namespace, other_client_id, other_client
 
   def _is_iap_host(self, host, client_id):
     if host and client_id:
-      if re.match(r'\S+.endpoints.\S+.cloud.goog/{0,1}$', host):
+      if host.endswith('/pipeline'):
```

Comment:
I think the problem here is how to detect whether the host is using IAP vs. inverse proxy.

If (perhaps by mistake), a user put `Client(host='1234567abc.pipelines.googleusercontent.com/#/pipeline')` then under this logic it will be determined as IAP, which is wrong.
I don't know whether we have a smarter way of doing that but am definitely happy to learn if there is one :)

Reply:
@IronPan and I had a quick sync earlier today. The proposal is to add an explicit parameter for user to specify whether it's IAP vs. inverse proxy. By default, when the type is unspecified, it falls back to the current detection logic. I'll put up a PR shortly.
@daikeshi what do you think about this approach?

Reply:
@chensun that works for us, thanks! I'll close this pr.

Reply:
just curious, does inverse proxy also require a client id? I originally thought `client_id` is only used by IAP. If `client_id` is provided, we can assume the host is an IAP endpoint, but having an explicit flag is probably better and less ambiguous.

Reply:
+1 to @daikeshi

Reply:
Thanks @daikeshi.
We had a quick sync today and client_id seems to be a good enough indicator. So we decide not to introduce another parameter for proxy type.
Made a PR: https://github.com/kubeflow/pipelines/pull/3003

Reply:
Sounds good! Thanks for the quick PR. I'm closing this one.